asmcomp/asmgen.ml

```diff
@@ -125,10 +125,19 @@ let compile_unit ~source_provenance asm_filename keep_asm obj_filename gen =
     raise exn
 
 let gen_implementation ?toplevel ~source_provenance ppf (size, lam) =
+  let main_module_block =
+    Clambda.{
```

Comment:
This syntax for local open on record is rather recent.  It's better to use the older syntax {Clambda.symbol = ...; ...}.